API/testrunner/devices/device_base.py

```diff
@@ -84,3 +84,28 @@ def logout(self):
         Logout from the device.
         '''
         self.channel.close()
+
+    def iotjs_build_info(self):
+        '''
+        Get buildinfo from iotjs.
+        '''
+        if self.device in ['rpi2', 'artik530']:
+            iotjs = '%s/iotjs' % self.workdir
+            buildinfo = '%s/tests/tools/iotjs_build_info.js' % self.workdir
+            command = ' '.join([iotjs, buildinfo])
+
+        elif self.device in ['artik053', 'stm32f4dis']:
+            command = 'iotjs /test/tools/iotjs_build_info.js'
+
+        self.login()
+
+        info_str = self.channel.exec_command(command)
+        info = json.loads(info_str[:info_str.rfind('}') + 1])
```

Comment:
Isn't enough a `json.loads(info_str)`? Or are there other strings after the JSON string?

Reply:
This is because on stm32 and artik053 there some extra information after iotjs finishes running, so the parts after the actual json string have to be removed. I will put a comment here to make it clear.